ProcessLib/HydroMechanics/HydroMechanicsFEM.h

```diff
@@ -99,9 +99,68 @@ struct IntegrationPointData final
     EIGEN_MAKE_ALIGNED_OPERATOR_NEW;
 };
 
+/// Used by for extrapolation of the integration point values. It is ordered
+/// (and stored) by integration points.
+template <typename ShapeMatrixType>
+struct SecondaryData
+{
+    std::vector<ShapeMatrixType, Eigen::aligned_allocator<ShapeMatrixType>> N;
```

Comment:
@TomFischer shape functions for the displacement variable are not directly stored in the integration point data; Would like to keep SecondaryData separate.